service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storage/StorageAccountKeyProvider.java

```diff
@@ -26,20 +30,36 @@ public StorageAccountKeyProvider(
     this.azureCloudContextService = azureCloudContextService1;
     this.crlService = crlService;
     this.azureConfiguration = azureConfiguration;
+    this.storageAccountKeyCache =
+        Collections.synchronizedMap(new PassiveExpiringMap<>(24, TimeUnit.HOURS));
```

Comment:
I stole this caching implementation from Jade's storage account [key caching impl](https://github.com/DataBiosphere/jade-data-repo/blob/cd537570e1d0fe8815b72591223b17401cb55008/src/main/java/bio/terra/service/resourcemanagement/azure/AzureAuthService.java#L156)